raftinc/allocate.hpp

```diff
@@ -45,89 +47,161 @@
 
 #define INITIAL_ALLOC_SIZE 64
 
-namespace raft
-{
-    class map;
-}
 
 class basic_parallel;
 
 class Allocate
 {
 public:
-   /**
-    * Allocate - base constructor, really doesn't do too much
-    * save for setting the global variables all_kernels and 
-    * source_kernels from the Map object.  
-    * @param map - raft::map&
-    * @param exit_alloc - bool used to terminate loop,
-    *  for monitoring allocations, controlled by map 
-    *  object.
-    */
-   Allocate( raft::map &map, volatile bool &exit_alloc );
-   
-   /**
-    * destructor 
-    */
-   virtual ~Allocate();
-
-   /**
-    * run - implement this function to create a new allocator, will
-    * be run inside a thread so exits when done but if run-time monitoring
-    * is desired then this is the place to do it.
-    */
-   virtual void run() = 0;
-
-   /** 
-    * waitTillReady - call after initializing the allocate thread, returns
-    * when the initial allocation is complete.
-    */
-   void waitTillReady() ;
-
-   
+    /**
+     * Allocate - base constructor, really doesn't do too much
+     * save for setting the global variables all_kernels and
+     * source_kernels from the Map object.
+     * @param map - raft::map&
+     * @param exit_alloc - bool used to terminate loop,
+     *  for monitoring allocations, controlled by map
+     *  object.
+     */
+    Allocate( MapBase &map, volatile bool &exit_alloc ) :
+        source_kernels( map.source_kernels ),
+        all_kernels( map.all_kernels ),
+        exit_alloc( exit_alloc )
+    {
+    }
+
+    /**
+     * destructor
+     */
+    virtual ~Allocate()
+    {
+        for( FIFO *f : allocated_fifo )
+        {
+#if RAFT_DUMP_STATS
```

Comment:
@jonathan-beard , find a better solution for this. Perhaps define a shared memory hook that is automatically dumped to if a telemetry agent is attached?